libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -427,6 +427,31 @@ const AP_Param::GroupInfo AP_GPS::var_info[] = {
 #endif // GPS_MAX_RECEIVERS > 1
 #endif // HAL_ENABLE_DRONECAN_DRIVERS
 
+    // @Param: _LOG_HZ
+    // @DisplayName: Log frequency
+    // @Description: The frequency with which the receiver logs information.
+    // @Units: Hz
+    // @Values: 0:Off, 1:0.1Hz, 2:0.2Hz, 5:0.5Hz, 10:1Hz, 20:2Hz, 50:5Hz, 80:8Hz, 100:10Hz, 200:20Hz
```

Comment:
The reasoning was to choose 1 for 0.1Hz as that is the lowest frequency a user would probably want. Choosing 1 for 1Hz would prevent adding a lower frequency in the future as there is no lower value than 1. I also tried to choose values that are available on common receivers.

Do you have a recommendation for how to change the parameter so it's more user friendly and generic? What should be the mapping between values (0.1Hz, 1Hz, 2Hz...) and underlying data (1, 10, 50...)? I would love any input on this approach.

Reply:
Just use floating point Hz.  And explain in the description that *the receiver will not guarantee that, it will only do a best effort to achieve it*

Reply:
Should the floating point values still be discrete (enum) or a continuous interval? For me it makes the most sense to use commonly available discrete values that are used on a best effort basis as most GNSS receivers use discrete frequency values for logging (100ms, 200ms, 1s...). A continuous interval could lead users to think that their receiver will use the exact value they choose, which could confuse them when they look at the actual value on their receiver.

Thank you for the help.

Reply:
Use a continuous floating point Hz. And explain in the description that the receiver will not guarantee that, it will only do a best effort to achieve it